text/_util.ts

```diff
@@ -1,9 +1,25 @@
 // Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+function splitByCase(input: string): string[] {
+  const acronym = /\p{Lu}{2,}/gu; // e.g. ID, URL
+  const capitalizedWord = /\p{Lu}\p{Ll}+/gu; // e.g. Apple
+  const lowercasedWord = /\P{Lu}+/gu; // e.g. apple, this will also match a sequence of unicode letters in languages without a concept of upper/lower case
+  const digits = /\p{N}+/gu; // e.g. 123
+
+  const matchWordByCase = new RegExp(
+    `${acronym.source}|${capitalizedWord.source}|${digits.source}|${lowercasedWord.source}`,
+    "gu",
+  );
+
+  const matches = input.match(matchWordByCase);
+
+  return matches || [input];
+}
 
 export function splitToWords(input: string) {
-  input = input.replaceAll(/[^a-zA-Z0-9\s-_]/g, "");
-  if (/[\s-_]+/.test(input)) return input.split(/[\s-_]+/);
-  return input.split(/(?=[A-Z])+/);
+  const words = input.split(/[^\p{L}\p{N}]+/gu).filter(Boolean).flatMap(
+    splitByCase,
+  );
+  return words;
```

Comment:
Nit: can you please also move the regex out into a variable?
```suggestion
  return input
    .split(/[^\p{L}\p{N}]+/gu)
    .filter(Boolean)
    .flatMap(splitByCase);
```